pkg/ee/kubelb/controller.go

```diff
@@ -199,67 +199,42 @@ func (r *reconciler) reconcile(ctx context.Context, cluster *kubermaticv1.Cluste
 	}
 
 	// Create/update required resources in user cluster.
-	if err := r.createOrUpdateKubeLBUserClusterResources(ctx, cluster); err != nil {
+	if err := r.createOrUpdateKubeLBUserClusterResources(ctx, cluster, datacenter); err != nil {
 		return nil, err
 	}
 
 	// Create/update required resources in user cluster namespace in seed.
-	if err := r.createOrUpdateKubeLBSeedClusterResources(ctx, cluster, kubeLBManagementClient, cfg, datacenter); err != nil {
+	result, err := r.createOrUpdateKubeLBSeedClusterResources(ctx, cluster, kubeLBManagementClient, datacenter)
+	if err != nil {
 		return nil, err
 	}
 
+	if result != nil {
+		return result, nil
+	}
+
 	return nil, nil
 }
 
 func (r *reconciler) createOrUpdateKubeLBManagementClusterResources(ctx context.Context, client ctrlruntimeclient.Client, cluster *kubermaticv1.Cluster) error {
-	namespace := fmt.Sprintf(kubelbresources.TenantNamespacePattern, cluster.Name)
-
-	// Create namespace; which is equivalent to registering the tenant.
-	nsReconcilers := []reconciling.NamedNamespaceReconcilerFactory{
-		kubelbmanagementresources.NamespaceReconciler(namespace),
-	}
-
-	if err := reconciling.ReconcileNamespaces(ctx, nsReconcilers, "", client); err != nil {
-		return fmt.Errorf("failed to reconcile namespace: %w", err)
-	}
-
-	// Create RBAC for the tenant.
-	saReconcilers := []reconciling.NamedServiceAccountReconcilerFactory{
-		kubelbmanagementresources.ServiceAccountReconciler(),
-	}
-
-	if err := reconciling.ReconcileServiceAccounts(ctx, saReconcilers, namespace, client); err != nil {
-		return fmt.Errorf("failed to reconcile service account: %w", err)
-	}
-
-	roleReconcilers := []reconciling.NamedRoleReconcilerFactory{
-		kubelbmanagementresources.RoleReconciler(),
-	}
-
-	if err := reconciling.ReconcileRoles(ctx, roleReconcilers, namespace, client); err != nil {
-		return fmt.Errorf("failed to reconcile role: %w", err)
-	}
-
-	roleBindingReconcilers := []reconciling.NamedRoleBindingReconcilerFactory{
-		kubelbmanagementresources.RoleBindingReconciler(namespace),
-	}
-
-	if err := reconciling.ReconcileRoleBindings(ctx, roleBindingReconcilers, namespace, client); err != nil {
-		return fmt.Errorf("failed to reconcile role binding: %w", err)
-	}
-
-	// Create service account token secret.
-	secretReconcilers := []reconciling.NamedSecretReconcilerFactory{
-		kubelbmanagementresources.SecretReconciler(),
+	// Since tenant configurations are not managed within KKP by this controller. We only check if tenant exists or not, if it doesn't we create it.
+	tenant := &kubelbv1alpha1.Tenant{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cluster.Name,
+		},
 	}
-
-	if err := reconciling.ReconcileSecrets(ctx, secretReconcilers, namespace, client); err != nil {
-		return fmt.Errorf("failed to reconcile secret: %w", err)
+	if err := client.Get(ctx, types.NamespacedName{Name: cluster.Name}, tenant); err != nil {
```

Comment:
For code stability, this could be `ctrlruntimeclient.ObjectKeyFromObject(tenant)`, then you don't have to repeat the "the tenant's name is the same as the cluster's name" logic (from line 223) here.